persistence/src/main/scala/zio/actors/persistence/EventSource.scala

```diff
@@ -0,0 +1,126 @@
+package zio.actors.persistence
+
+import java.io.File
+
+import zio.actors.{ Actor, Context, Supervisor }
+import zio.{ IO, Queue, RIO, Ref, Task, ZIO }
+import zio.actors.Actor._
+import zio.actors.persistence.Utils.JournalPlugin
+import zio.actors.persistence.journal.{ InMemJournal, JDBCJournal, Journal }
+
+/**
+ *
+ * Each message can result in either an event that will be persisted or idempotent action.
+ * Changing the actor's state can only occur via `Persist` event.
+ *
+ * @tparam Ev events that will be persisted
+ */
+sealed trait Command[+Ev]
```

Comment:
Can you re-organize the module to have one top-level artifact? Also, try to put ADT subtypes into their companions.

Reply:
Will put them into companions.

I'm not sure about the first one. The module produces single maven artifact: `zio-actors-persistence`. Could you give me some hint on that?

Reply:
No, no, by module I didn't mean sbt project, rather `EventSource.scala`, sorry for misleading. Re: deliverables, we can reorganize that a bit later.